File: app/src/main/java/com/mauriciotogneri/greencoffeeexample/adapters/ContactViewHolder.java
package com.mauriciotogneri.greencoffeeexample.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.mauriciotogneri.greencoffeeexample.R;
import com.mauriciotogneri.greencoffeeexample.model.Contact;

public class ContactViewHolder extends RecyclerView.ViewHolder {
    private TextView firstName;
    private TextView age;
    private TextView weight;
    private TextView married;

    public ContactViewHolder(@NonNull View view) {
        super(view);

        firstName = view.findViewById(R.id.contact_name);
        age = view.findViewById(R.id.contact_age);
        weight = view.findViewById(R.id.contact_weight);
        married = view.findViewById(R.id.contact_married);
    }

    public void bind(Contact contact) {
        itemView.setTag(contact);
        firstName.setText(contact.name());
        age.setText(contact.age());
        weight.setText(contact.weight());
        married.setText(contact.married(itemView.getContext()));
    }
}

File: app/src/main/java/com/mauriciotogneri/greencoffeeexample/adapters/ContactViewHolder.kt
package com.mauriciotogneri.greencoffeeexample.adapters

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import com.mauriciotogneri.greencoffeeexample.R
import com.mauriciotogneri.greencoffeeexample.model.Contact

// 类成员声明时就可以用构造函数的参数进行初始化，简洁
class ContactViewHolder(view: View) : RecyclerView.ViewHolder(view) {
    private val firstName: TextView = view.findViewById(R.id.contact_name)
    private val age: TextView = view.findViewById(R.id.contact_age)
    private val weight: TextView = view.findViewById(R.id.contact_weight)
    private val married: TextView = view.findViewById(R.id.contact_married)

    fun bind(contact: Contact) {
        itemView.tag = contact
        firstName.text = contact.name()
        age.text = contact.age()
        weight.text = contact.weight()
        married.text = contact.married(itemView.context)
    }
}
